herddb-core/src/test/java/herddb/core/CreateTableTest.java

```diff
@@ -63,4 +63,50 @@ public void createTable1() throws Exception {
         }
 
     }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testWithNullPrimaryKey() throws Exception {
+        String nodeId = "localhost";
+        try (DBManager manager = new DBManager("localhost", new MemoryMetadataStorageManager(), new MemoryDataStorageManager(), new MemoryCommitLogManager(),null, null);) {
+            manager.start();
+            CreateTableSpaceStatement st1 = new CreateTableSpaceStatement("tblspace1", Collections.singleton(nodeId), nodeId, 1, 0, 0);
+            manager.executeStatement(st1, StatementEvaluationContext.DEFAULT_EVALUATION_CONTEXT(), TransactionContext.NO_TRANSACTION);
+            manager.waitForTablespace("tblspace1", 10000);
+
+            Table table = Table
+                    .builder()
+                    .tablespace("tblspace1")
+                    .name("t1")
+                    .column("id", ColumnTypes.DOUBLE)
+                    .column("name", ColumnTypes.STRING)
+                    .primaryKey("id")
+                    .build();
+
+            CreateTableStatement st2 = new CreateTableStatement(table);
+            manager.executeStatement(st2, StatementEvaluationContext.DEFAULT_EVALUATION_CONTEXT(), TransactionContext.NO_TRANSACTION);
+        }
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testWithNullPrimaryKey2() throws Exception {
```

Comment:
same as above

Reply:
This test is to ensure we throw exception when primary key is chosen as ByteArray

Reply:
please rename

Reply:
Done.